c2/c2-protocol/c2-protocol-api/pom.xml

```diff
@@ -33,5 +33,9 @@ limitations under the License.
             <artifactId>c2-protocol-component-api</artifactId>
             <version>1.18.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
```

Comment:
I think we should avoid polluting the -api modules. They should be pure only with the models (maybe it will be used with different serialisation not Jackson somewhere). Could you look into a different way of addressing the issue this solves? We can discuss separately if needed.

Reply:
The reason behind this change is that different clients sends the orderType values in different ways(upper or lower case). Before adding the @JsonCreatorit annotation, I checked and it is used several places in nifi-api modules. I can have the same result by writing a Custom Deserializer and add it to an ObjectMapper in C2JacksonSerializer. I think it's a more complicated solution and I don't think that there is any problem with adding the jackson dependency to an api module. Please let me know, if you still insist to change it, I'll do that.

Reply:
I understand the reason but we have to keep the api module clean as this is basically the c2 protocol domain definition which shouldn't be opinionated (if someone is using the protocol but uses different things for serialisation they shouldn't depend on jackson). So please remove it from here.

Reply:
Concurring with @bejancsaba, although the Jackson annotations are convenient, the `c2-protocol-api` dependencies should be kept at a minimum, so writing a custom deserializer would be preferred in this particular case.

Reply:
Removed the dependency and added a custom deserializer.

Reply:
Thank you!